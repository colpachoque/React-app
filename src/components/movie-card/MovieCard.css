.movieCard {
    margin: 50px;
}

.movieImg {
    width: 150px;
    height: 200px;
    background-image: url('../../../resources/killBill.png');
    background-repeat: no-repeat;
    background-size: cover;
}

.movieInfo {
    width: 150px;
    display: flex;
    flex-wrap: wrap;
}

.name {
    font-weight: 600;
    text-transform: uppercase;
    width: 50%;
}

.genre {
    width: 100%;
    font-size: small;
}

.year {
    text-align: right;
    width: 50%;
}
