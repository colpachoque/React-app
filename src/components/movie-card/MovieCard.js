import React from "react";
import styles from "./MovieCard.css";
import { HashRouter as Link, NavLink } from "react-router-dom";
class MovieCard extends React.Component {
  render() {
    return (
      <div className={styles.movieCard}>
        <div className={styles.movieImg} />
        <div className={styles.movieInfo}>
          <div className={styles.name}>{this.props.name}</div>
          <div className={styles.year}>{this.props.year}</div>
          <div className={styles.genre}>{this.props.genre.join(", ")}</div>
        </div>
        <NavLink to={`/movie/${this.props.movieId}`}>Details</NavLink>
      </div>
    );
  }
}

export default MovieCard;
