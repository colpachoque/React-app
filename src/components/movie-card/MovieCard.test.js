import React from "react";
import MovieCard from "./MovieCard";
import { shallow, mount } from "enzyme";

describe("Movie Card component", () => {
  let movie;

  beforeAll(() => {
    movie = {
      name: "The best movie ever",
      genre: "Drama",
      year: 1994
    };
  });

  it("should be render correctly", () => {
    const component = shallow(
      <MovieCard name={movie.name} genre={movie.genre} year={movie.year} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render correct year", () => {
    const component = mount(
      <MovieCard name={movie.name} genre={movie.genre} year={movie.year} />
    );
    expect(component.prop("year")).toEqual(1994);
  });

  it("should render correct name", () => {
    const component = mount(
      <MovieCard name={movie.name} genre={movie.genre} year={movie.year} />
    );
    expect(component.props().name).toBeDefined();
  });

  it("should render correct genre", () => {
    const component = mount(
      <MovieCard name={movie.name} genre="Drama" year={movie.year} />
    );
    expect(component.prop("genre")).toEqual(movie.genre);
  });
});
