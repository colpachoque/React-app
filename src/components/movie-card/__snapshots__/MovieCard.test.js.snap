// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card component should be render correctly 1`] = `
<div
  className="movieCard"
>
  <div
    className="movieImg"
  />
  <div
    className="movieInfo"
  >
    <div
      className="name"
    >
      The best movie ever
    </div>
    <div
      className="year"
    >
      1994
    </div>
    <div
      className="genre"
    >
      Drama
    </div>
  </div>
</div>
`;
